Corrfunc/mocks/DDtheta_mocks.py

```diff
@@ -351,6 +351,304 @@ def DDtheta_mocks(autocorr, nthreads, binfile,
+
+    integer_isa = translate_isa_string_to_enum(isa)
+    rbinfile, delete_after_use = return_file_with_rbins(binfile)
+    bin_refs = np.arange(1, maxbinref + 1)
+    if link_in_ra:
+        bin_ref_perms = itertools.product(bin_refs, bin_refs)
+    else:
+        bin_ref_perms = [(1, binref) for binref in bin_refs]
+    dtype = np.dtype([(bytes_to_native_str(b'nRA'), np.int),
+                      (bytes_to_native_str(b'nDEC'), np.int),
+                      (bytes_to_native_str(b'avg_time'), np.float),
+                      (bytes_to_native_str(b'sigma_time'), np.float)])
+    all_runtimes = np.zeros(maxbinref ** 3, dtype=dtype)
+    all_runtimes[:] = np.inf
+
+    if autocorr == 0:
+        if RA2 is None or DEC2 is None:
+            msg1 = "Must pass valid arrays for RA2/DEC2 for "\
+                  "computing cross-correlation."
+            raise ValueError(msg1)
+    else:
+        RA2 = np.empty(1)
+        DEC2 = np.empty(1)
+
+    if link_in_ra is True:
+        link_in_dec = True
+
+    if link_in_dec is False and link_in_ra is False:
+        msg2 = "Warning: Brute-force calculation without any gridding "\
+               "is forced, as link_in_dec and link_in_ra are both set "\
+               "to False. Please be sure to turn on at least link_in_dec, "\
+               "or both link_in_dec and link_in_ra on."
+        raise ValueError(msg2)
+
+    if link_in_ra:
+        pass
+
+    if link_in_dec is True:
+        if link_in_ra is False:
+            msg3 = "Info: Gridding in the declination only, as link_in_dec "\
+                   "is set to True while link_in_ra is set to False. "\
+                   "Thus looping is only needed over the range of "\
+                   "(min, max) bin, with refinements in the declination."
+            print(msg3)
+
+    for ii, (nRA, nDEC) in enumerate(bin_ref_perms):
+        total_runtime = 0.0
+        total_sqr_runtime = 0.0
+
+        for _ in range(nrepeats):
+            t0 = time.time()
+            extn_results = DDtheta_mocks_extn(autocorr, nthreads, rbinfile,
+                                              RA1, DEC1, RA2, DEC2,
+                                              link_in_dec=link_in_dec,
+                                              link_in_ra=link_in_ra,
+                                              verbose=verbose,
+                                              output_thetaavg=output_thetaavg,
+                                              ra_refine_factor=nRA,
+                                              dec_refine_factor=nDEC,
+                                              max_cells_per_dim=max_cells_per_dim,
+                                              isa=integer_isa)
+            t1 = time.time()
+
+            if extn_results is None:
+                msg4 = "RuntimeError occurred with perms = ({0}, {1})".\
+                    format(nRA, nDEC)
+                print(msg4)
+                print("Continuing...")
+                continue
```

Comment:
Hmm - you should raise `ValueError(msg)` here. If the underlying code crashed, it is unlikely that the next iteration will succeed. Probably better to let the user debug what might be going on